Rephrase documents to gender-sensitive language

hochleitner opened the pull request:
This PR contains a few changes, the most important being that all German template documents (mostly the thesis template) now adhere to gender-sensitive language using the asterisk (\*) notation. Here are the changes in detail:

- All mentions of persons, responsibilities, or activities that were phrased in the masculine form (e.g., Autor) has been replaced with a form that included the whole gender-spectrum (e.g., der\*die Autor\*in).
- Example names, such as the one of the fictional student "Peter A. Schlaumeier," have been switched to a neutral form. Alex was chosen since it works in every direction.
- References were updated. Some outdated ones were removed (e.g., the old Master's thesis example from Digitale Medien since the URL to the PDF was no longer active). Others were updated (mostly the LaTeX package manuals, also a book).
- A few typos and punctuations were fixed, and links were updated.
- Submission dates were forwarded to 2021.
- Most overfull boxes were corrected.

@imagingbook, have a look at it and see if you find the changes okay. There's no rebuild included, just the bare changes in the tex and bib files.

I also think this would be a good time to publish a new release. The unreleased section in the changelog is rather long and full of cool new stuff.

imagingbook commented:
Thanks for the effort, looks good! However, I see a typographic problem in using a simple '*' in the gendered expressions, since in LaTeX this character is not meant for this purpose and is rendered as a superscripted asterisc. The proper substitute would be ``\textasteriskcentered`` or ``$\ast$`` (the latter being a kludge), which is probably unacceptable to authors.
One solution could be to re-define LaTeX's interpretation of the '*' character, which I prefer to avoid for unknown consequences (macro interpretation etc.).
What I suggest instead is to provide a set of LaTeX macros for gendering in German instead. E.g., instead of ``Benutzer{\textasteriskcentered}innen`` one would simply write ``\gender{Benutzer}{innen}`` or such. We could also define macros for frequent expressions like ``den*die``, ``der*die`` etc. This way we could also make proper provisions for undisturbed line breaking. All this packaged into a neat style file, of course ...

What dou you think?

hochleitner commented:
I have been thinking about this as well, but I actually discarded the thoughts of using something like `\textasteriskcentered` again because it's way too complicated. Plus, the asterisk has not been standardized in any way right now, so there is no definitive character that has been assigned. It all originated from a technical perspective where `*.txt` would mean "all text files." This was ported to the gender domain, where * would mean "everything between man and woman." The asterisk in `*.txt` is superscripted after all, so I don't think that it is necessarily wrong.

I think all of this is still in flux. We should encourage people (mostly students) to actively participate in the discourse by setting positive examples and including all genders. Thus we should make it as easy as possible for them to work with the asterisk.

I totally forgot to write a section about our reasoning for this. I will add this to the PR (please don't merge just yet). I can add that from a typographical point of view, another asterisk might be more appropriate, but we went with simplicity until this matter has been standardized.

imagingbook commented:
I get the point and we should week a solution that makes this least invasive for authors. I consider the line breaking issue more important than the raised * problem. Are there any known standards yet?
Also, I am surprised (considering how long this topic has been around) that nobody in the (German) LaTeX community has made a useful proposal/package yet. 
Let me play a little, perhaps there IS a simpler solution ...

hochleitner commented:
> I get the point, and we should seek a solution that makes this least invasive for authors. I consider the line breaking issue more important than the raised * problem. Are there any known standards yet?

I've been looking through many documents, and all of them use the regular asterisk (e.g., https://gb.uni-koeln.de/e2106/e2113/e16894/2019_Leitfaden_GendergerechteSprache_19022020_32_Poster_Webausgabe_ger.pdf). One might say that these documents have not been created with typographic conventions in mind (which is probably correct). Still, as I stated before, simplicity is key here.
The asterisk is a de-facto standard, but no explicit character has been defined yet since no real standard organization has picked it up. Judging from how it is commonly used, it's definitely the regular asterisk (\*) and not the centered asterisk (✲)

> Also, I am surprised (considering how long this topic has been around) that nobody in the (German) LaTeX community has made a useful proposal/package yet.

Maybe because everyone's just using the regular asterisk? 

> Let me play a little, perhaps there IS a simpler solution ...

Sure, but maybe there's just no issue here at all. Maybe it's just the plain old, superscripted asterisk: https://www.dwds.de/wb/Gender-Sternchen

imagingbook commented:
OK, so let's stick with the raised asterix (oddly, ``\textasteriskcentered`` does not render cleanly on my system anyways).
However, after a few quick tests it is clear that line-braking will be a disaster. This needs to be fixed somehow. So, where to break `Nutzer*innen` -- before or after the *? Any rules by those smart people?

hochleitner commented:
Well, hyphenation isn't worse than German words with actual hyphens. But there are a few possibilities; they are listed here: https://geschicktgendern.de/gendersternchen-anwenden/

I'll add that to my gender-sensitive language section.

imagingbook commented:
> Well, hyphenation isn't worse than German words with actual hyphens. 

No, it definitely IS MUCH WORSE, sorry...

> But there are a few possibilities; they are listed here: https://geschicktgendern.de/gendersternchen-anwenden/

This lists practically all possiblities (including some really idiotic ones, e.g. "avoid hyphenation at all"), so I guess there are NO rules. Makes me sick -- why can't such (admittedly difficult) things not be done right from the beginning?
I suggest to make it either (a) `Nutzer- *innen` or (b) `Nutzer* -innen`. Both version can be automated, I think. 
Which one should we go for?

hochleitner commented:
> I suggest to make it either (a) `Nutzer- *innen` or (b) `Nutzer* -innen`. Both version can be automated, I think.
> Which one should we go for?

I delegated this question to a linguist who's very active in the field of gender-sensitive language: https://twitter.com/GangMan_/status/1321087267319013390.
Let's see what the expert says. :)

hochleitner commented:
Well, no reply so far. I'll add it here if I hear anything back. I also sent this question to the team of https://www.genderleicht.de/. They have a "Textlabor" with some examples when it comes to gender-sensitive language.

Intuitively I'd go with `Nutzer* -innen` by the way.

hochleitner commented:
@imagingbook: I added a section about language and writing style to the "Abschlussarbeit" section. Have a look at it and tell me if you want more details or change as you see fit. I also added this to the checklist at the end of the document.

imagingbook commented:
@hochleitner Looks good! I will do a full rebuild/release asap.

imagingbook commented:
I checked around what LaTeX's hyphenation is handling *-"enriched" words in German. Here are  examples taken from our tutorial (using package ``testhyphens``):
<pre>
der*-die
sei-ne*ih-re
ei-nem*ei-ner
er*-sie
des*-der
je-de*r
Nut-zer*in-nen
An-fän-ger*in-nen
Lai-in*Lai-en
Pu-rist*in-nen
den*-die
Le-ser*in
Au-tor*in
Au-tor*in-nen
Stu-dent*in-nen
Teil-neh-mer*in-nen
Buch-bin-der*in
Au-tor-s*-Au-to-rin
Be-nut-zer*in-nen
Au-tor*in-ne-num-ge-bung
Be-treu-er-s*-Be-treue-rin
Be-treu-er*in-nen-sei-te
Her-aus-ge-ber-s*Her-aus-ge-be-rin
Kol-leg*in-nen
Be-gut-ach-ter*in
</pre>
It appears that the default behavior is not as rigid as I though but actually quite acceptable. So I suggest to be happy and not go for any additional measures. It would be very difficult anyways to modify the hyphenation patterns in general.

imagingbook commented:
@hochleitner Rebuild is complete and pushed. I will update the CTAN-submission -- do you want to do the GIT release?

hochleitner commented:
Sure, I'll create a tag, a release and update the release notes accordingly.

imagingbook commented:
> Sure, I'll create a tag, a release and update the release notes accordingly.

Pls. wait, there was a problem with the build script - need to clean up first!

imagingbook commented:
OK, all fine now. `latexmake` did not clean up as intended, I fixed the makefile(s).
